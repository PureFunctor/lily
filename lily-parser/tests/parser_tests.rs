use lily_parser::parse_top_level;

#[test]
pub fn top_level_0() {
    let source = r"
infixl 4 add as +

example = a + b + c
";

    insta::assert_debug_snapshot!(parse_top_level(source));
}

#[test]
pub fn top_level_1() {
    let source = r"
example = a + b + c
";

    insta::assert_debug_snapshot!(parse_top_level(source));
}
