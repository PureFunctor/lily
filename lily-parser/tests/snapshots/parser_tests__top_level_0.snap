---
source: lily-parser/tests/parser_tests.rs
expression: parse_top_level(source)
---
Ok(
    Module {
        declarations: [
            Declaration {
                begin: 20,
                end: 39,
                kind: ValueDeclaration(
                    "example",
                    [],
                    Expression {
                        begin: 30,
                        end: 39,
                        kind: BinaryOperator(
                            Expression {
                                begin: 30,
                                end: 35,
                                kind: BinaryOperator(
                                    Expression {
                                        begin: 30,
                                        end: 31,
                                        kind: Variable(
                                            "a",
                                        ),
                                    },
                                    "+",
                                    Expression {
                                        begin: 34,
                                        end: 35,
                                        kind: Variable(
                                            "b",
                                        ),
                                    },
                                ),
                            },
                            "+",
                            Expression {
                                begin: 38,
                                end: 39,
                                kind: Variable(
                                    "c",
                                ),
                            },
                        ),
                    },
                ),
            },
        ],
    },
)
